src/app/routes/applications/new-role/new-role.component.ts

```diff
@@ -465,6 +476,23 @@ export class NewRoleComponent implements OnInit, AfterViewInit, OnDestroy {
     this.dataSource.data = [...list];
   }
 
+  editField(i: number) {
+    this.fieldIndex = i;
+    let field = this.dataSource.data[i];
```

Comment:
It's a small thing but use const where possible instead of let. We didn't set up tslint/eslint in travis, so this needs to be handled manually :(